File: src/lessons/welcome/slug/SlugHuntingEntity.py
import java.awt.Color as Color

# BEGIN TEMPLATE
# BEGIN HIDDEN
def isFacingTrail():
   if isFacingWall():
      return False
   else:
	  forward()
	  res = getGroundColor().toString() == Color.green.toString()
	  backward()
	  return res
# END HIDDEN

def hunt():
   # write your code here
   # BEGIN SOLUTION
   while not isOverBaggle():
      brushUp()
      if isFacingTrail():
         brushDown()
         forward()
         brushUp()
      else:
	     turnLeft()
   pickupBaggle()
   # END SOLUTION
# END TEMPLATE

hunt()

File: src/lessons/welcome/methods/slug/SlugHuntingEntity.scala
package lessons.welcome.methods.slug;

import java.awt.Color;

class SlugHuntingEntity extends plm.universe.bugglequest.SimpleBuggle {

	override def run() {
		hunt(); 
	}

	/* BEGIN TEMPLATE */
	def hunt() {
		// Write your code here
		/* BEGIN SOLUTION */
		while (! isOverBaggle()) {
			if (isFacingTrail()) {
				brushDown();
				stepForward();
				brushUp();
			} else {
				left();
			}
		}
		pickupBaggle();
	}
	
	def isFacingTrail():Boolean = {
		if (isFacingWall())
			return false;

		stepForward();
		var res = getGroundColor() == Color.green;
		stepBackward();
		return res;

		/* END SOLUTION */
	}
	
	// Copy your isFacingTrail() here
	/* END TEMPLATE */
}
